src/Github/MergeMultipleReleaseNotes.php

```diff
@@ -33,19 +32,24 @@ public function __invoke(
         SemVerVersion $semVerVersion,
         BranchName $sourceBranch,
         string $repositoryDirectory
-    ): string {
+    ): ChangelogReleaseNotes {
         $items = array_map(
-            static fn (CreateReleaseText $generator): string => $generator($milestone, $repositoryName, $semVerVersion, $sourceBranch, $repositoryDirectory),
+            static fn (CreateReleaseText $generator): ChangelogReleaseNotes => $generator($milestone, $repositoryName, $semVerVersion, $sourceBranch, $repositoryDirectory),
             array_filter(
                 $this->releaseTextGenerators,
                 static fn (CreateReleaseText $generator): bool => $generator->canCreateReleaseText($milestone, $repositoryName, $semVerVersion, $sourceBranch, $repositoryDirectory)
             )
         );
 
-        $changelog = implode(self::CONCATENATION_STRING, $items);
-        Assert::notEmpty($changelog);
+        $releaseNotes = array_reduce(
+            $items,
+            static fn (?ChangelogReleaseNotes $releaseNotes, ChangelogReleaseNotes $item): ChangelogReleaseNotes => $releaseNotes ? $releaseNotes->merge($item) : $item
+        );
+
+        Assert::isInstanceOf($releaseNotes, ChangelogReleaseNotes::class);
+        Assert::notEmpty($releaseNotes->contents());
```

Comment:
Is this assertion needed? I think `->contents()` already returns a `non-empty-string`